### PR TITLE
offset2D implementation

### LINKED ISSUE #208: Question: offset curve / tolerance
I want to expand a 2D polyline surface by 1mm.
Similar to https://forum.freecadweb.org/viewtopic.php?t=9137

Or, is it on the roadmap?
https://cadquery.readthedocs.io/en/latest/roadmap.html#d-operations

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Will resolve #208 and #347 

- [x] Implementation
- [x] Docs

obraz

### CODE REVIEW

File: cadquery/cq.py [hunk 1]
```diff
@@ -1933,44 +1934,6 @@ def tangentArcPoint(
 
         return self.newObject([arc])
 
-    '''
-    def rotateAndCopy(self, matrix: Matrix) -> "Workplane":
```
Comment (jmwright) [inline cadquery/cq.py:1937]: Why was this commented out and now removed?

File: cadquery/cq.py [hunk 2]
```diff
@@ -3675,6 +3648,26 @@ def toPending(self) -> "Workplane":
 
         return self
 
+    def offset2D(
+        self, d: float, kind: Literal["arc", "intersection", "tangent"] = "arc"
+    ) -> "Workplane":
+        """
+        Creates a 2D offset wire.
+        
+        :param float d: thickness. Negative thickness denotes offset to inside.
```
Comment (jmwright) [inline cadquery/cq.py:3657]: So this implies that a 0 offset can be used, even though it's not practical to do so. Does the kernel handle that properly?

File: tests/test_cadquery.py [hunk 3]
```diff
@@ -2771,7 +2770,7 @@ def testMakeShellSolid(self):
                 lines.append(
                     Edge.makeLine(Vector(*vertices[v1]), Vector(*vertices[v2]))
                 )
-            wire = Wire.combine(lines)
+            wire = Wire.combine(lines)[0]
```
Comment (jmwright) [inline tests/test_cadquery.py:2773]: Why is only the first wire used here after the combine?

File: cadquery/cq.py (hunk 1 above)
Comment (adam-urbanczyk) [reply cadquery/cq.py:1937]: I just removed commented code. Looks like I might have commented it our by mistake in previous PRs - I'll revert this change.

File: cadquery/cq.py (hunk 2 above)
Comment (adam-urbanczyk) [reply cadquery/cq.py:3657]: 0 offset just works

File: tests/test_cadquery.py (hunk 3 above)
Comment (adam-urbanczyk) [reply tests/test_cadquery.py:2773]: I changed Wire.combine to return a list of Wires. Now it handles disconnected edges/wires properly, but obviously such a modification is needed.

File: cadquery/cq.py (hunk 1 above)
Comment (adam-urbanczyk) [reply cadquery/cq.py:1937]: I did some brief archeology and now it is clear why I commented it out: it was calling the method `rotateShapes` which was not implemented (i.e. throwing a `NotImplementedError`). Hence I removed both.

**Comment by jmwright** [comment]
Ok. I think all discussion items have been resolved. Merging.

**Pull request merged**